grug/app/src/app.rs

```diff
@@ -190,6 +190,14 @@ where
             .prepare_proposal(QuerierWrapper::new(&querier), txs, max_tx_bytes)?)
     }
 
+    // Finalize a block by performing the following actions in order:
+    //
+    // 1. indexer `pre_indexing`
+    // 2. execute transactions one by one
+    // 3. perform cronjobs
+    // 4. remove orphaned nodes
+    // 5. flush (but not commit) state changes to DB
+    // 5. indexer `index_block`
```

Comment:
move cronjobs after transactions. this way, if user submits an order, and there's a match, then the order is filled **in the same block** where it's submitted. otherwise, user has to wait until the next block for the order to be filled.